larana/OpticalDetector/SimPhotonCounter_module.cc

```diff
@@ -280,39 +295,34 @@ namespace opdet {
     art::EventNumber_t event = evt.id().event();
     fEventID=Int_t(event);
 
-    art::ServiceHandle<phot::PhotonVisibilityService const> pvs;
-
-    art::ServiceHandle<sim::LArG4Parameters const> lgp;
-    bool fUseLitePhotons = lgp->UseLitePhotons();
-
     // Service for determining opdet responses
     art::ServiceHandle<opdet::OpDetResponseInterface const> odresponse;
 
     // get the geometry to be able to figure out signal types and chan -> plane mappings
     art::ServiceHandle<geo::Geometry const> geo;
 
     // get the MCtrue info of the particles
-    art::Handle< std::vector<simb::MCParticle> > mclistLARG4;
-    evt.getByLabel("largeant",mclistLARG4);
-    std::vector<simb::MCParticle> const& mcpartVec(*mclistLARG4);
+    std::vector<simb::MCParticle> const* mcpartVec = fLightAnalysisTree
+      ? evt.getPointerByLabel<std::vector<simb::MCParticle>>("largeant")
+      : nullptr;
```

Comment:
This is a very awkward construct.  I recommend refactoring the code so you do not risk an unintentional dereference of `nullptr`.

Reply:
See next answer...